src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```
@@ -0,0 +1,133 @@
+               storage_account_key=None, deployment_name="Microsoft.ContainerRegistry"):
+    '''Returns the created container registry.
+    :param str resource_group: The name of resource group
+    :param str registry_name: The name of container registry
+    :param str location: The name of location
+    :param str storage_account_name: The name of storage account
+    :param str storage_account_key: The key of storage account
+    :param str deployment_name: The name of deployment
+    '''
+    validate_registry_name(registry_name)
+    if storage_account_name:
+        if storage_account_key:
+            storage_account_properties = \
+            StorageAccountProperties(name=storage_account_name,
+                                     access_key=storage_account_key,
+                                     endpoint_suffix=get_storage_end_point_suffix())
+            registry_properties = \
+            RegistryProperties(storage_account=storage_account_properties)
+            registry_parameters = \
+            RegistryParameters(location=location,
+                               properties=registry_properties)
+            return get_registry_service_client().create(
+                resource_group, registry_name, registry_parameters)
+        else:
+            return arm_deploy_template(
+                resource_group, registry_name, location, storage_account_name, deployment_name)
+    else:
+        return get_registry_service_client().create(
+            resource_group, registry_name, RegistryParameters(location=location))
+
+def acr_delete(registry_name):
+    '''Deletes the container registry that matches the registry name.
+    :param str registry_name: The name of container registry
+    '''
+    validate_registry_name(registry_name)
+    registry = arm_get_registry_by_name(registry_name)
+    if registry is None:
+        logger.error('No container registry can be found with name: ' + registry_name)
```

raise CLIError instead of following this pattern of logger.error then raise SystemExit.